PVT Tables logging improvements

arng40 opened the pull request:
This PR follows the #2984 that aim to standardize logs.
Currently PVT tables are only accessible throught .csv files.
In this PR, logs will indicate which csv files are generated along with their associate path
``` 
CSV Generated to inputFiles/compositionalMultiphaseWell/test.xml/fluid1_phaseModel1_PhillipsBrineDensity_table.csv :
---------------------------------------------------------------------------------------
|                    fluid1_phaseModel1_PhillipsBrineDensity_table                    |
---------------------------------------------------------------------------------------
|  The fluid1_phaseModel1_PhillipsBrineDensity_table PVT table exceeding 500 rows.    |
|  To visualize the tables, go to the generated csv                                   |
---------------------------------------------------------------------------------------
```
In addition to csv files, PVT Tables with dimensions n <= 2 can be written in log by meetings 2 conditions : 
- have sufficient log lovel
- Rows in the table < 500 lines

```
CSV Generated to inputFiles/compositionalMultiphaseWell/test.xml/fluid1_phaseModel1_PhillipsBrineViscosity_table.csv
---------------------------------------------------------
|    fluid1_phaseModel1_PhillipsBrineViscosity_table    |
---------------------------------------------------------
|    temperature [C]     |       viscosity [Pa*s]       |
---------------------------------------------------------
|          0.01          |          0.0017914           |
|           10           |           0.001306           |
|           20           |          0.0010016           |
|           25           |           0.00089            |
|           30           |          0.0007972           |
---------------------------------------------------------

arng40 committed:
Table function creation

arng40 committed:
add title off case

arng40 committed:
setMargin function added

arng40 committed:
add comments + test for tiny table

arng40 committed:
value default optionnal variable changed

arng40 committed:
Merge remote-tracking branch 'origin/develop' into feature/dudes/table-layout

arng40 committed:
add log to wellGeneratorBase

arng40 committed:
uncrustify + missing doc

arng40 committed:
first review with correction

arng40 committed:
constness modification, doc/test updated

arng40 committed:
const declaration updated

arng40 committed:
doc + minor restructuring

arng40 committed:
doc + method for row from vector

arng40 committed:
clean code + doc + add test for rows from vector

arng40 committed:
begin pvt tables logs - add write in log bool

arng40 committed:
doc updated

arng40 committed:
duplication log & csv fixed

arng40 committed:
Merge commit 'c11ad8ecd2d0d0aba5cd0661eb6e72951252409d' into feature/dudes/table-layout

arng40 committed:
Revert "Merge commit 'c11ad8ecd2d0d0aba5cd0661eb6e72951252409d' into feature/dudes/table-layout"

This reverts commit 8f74cfa7027ab13a3b82f0be10ba83fccf1f04ba.

arng40 committed:
revert last merge and Merge remote-tracking branch 'origin/develop' into feature/dudes/table-layout

arng40 committed:
Merge remote-tracking branch 'origin/feature/dudes/table-layout' into pvt-logs-cp

arng40 committed:
CI correction ( uncrustify, missing doc )

arng40 committed:
optionnal added to datatype

arng40 committed:
update doxygen for ci

arng40 committed:
more logs added

arng40 committed:
remove static_assert + constness

arng40 committed:
Merge remote-tracking branch 'origin/develop' into feature/dudes/table-layout

MelReyCG commented on src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/BrineEnthalpy.cpp:
```diff
@@ -204,11 +205,18 @@ BrineEnthalpy::BrineEnthalpy( string const & name,
-    m_brineEnthalpyTable->print( m_brineEnthalpyTable->getName() );
+    m_CO2EnthalpyTable->printInLog( m_CO2EnthalpyTable->getName() );
+    m_brineEnthalpyTable->printInLog( m_brineEnthalpyTable->getName() );
   }
```
Here, you are assuming that `m_CO2EnthalpyTable` and `m_brineEnthalpyTable` have the same dimensions. `m_CO2EnthalpyTable` is "choosing" the formatting for `m_brineEnthalpyTable`.
But this assomption is wrong, they does not have the same layout !

You could do two `if` for each tables, or you could create a function that factorize this choice, something like `TableFunction::print( bool printInCsv, bool printInLog )`.
Factorizing can be a good thing as there we need 13 of these conditions.

MelReyCG commented on src/coreComponents/constitutive/fluid/multifluid/CO2Brine/CO2BrineFluid.cpp:
```diff
@@ -231,13 +237,27 @@ void CO2BrineFluid< PHASE1, PHASE2, FLASH >::postProcessInput()
+    std::cout << sectionOutput;
+
+    isClone = false;
+  }
```
Can we do this job without this if ? The relation with the "Constitutive" parent is external to this class. If possible, I think that doing with your `isClone` parameter is clearer and more maintainable.

MelReyCG reviewed (changes requested):
Some refactor suggestions

MelReyCG commented on src/coreComponents/constitutive/fluid/multifluid/CO2Brine/CO2BrineFluid.cpp:
```diff
@@ -231,13 +237,27 @@ void CO2BrineFluid< PHASE1, PHASE2, FLASH >::postProcessInput()
+  if( getParent().getName() == "Constitutive" )
+  {
+    string sectionOutput;
+    string sectionName = "Section : PVT Table generation";
+
+    sectionOutput = GEOS_FMT( "\n{:=^{}}\n", "=", sectionName.size() + 5 );
+    sectionOutput += GEOS_FMT( "{:^{}}\n", sectionName, sectionName.size() + 5 );
+    sectionOutput += GEOS_FMT( "{:=^{}}\n\n", "=", sectionName.size() + 5 );
+
+    std::cout << sectionOutput;
```
Just a reminder to keep that in mind, those sections layout will need to be factorized in a dedicated PR.

arng40 committed:
remove unused code

arng40 committed:
factorisation log pvt

arng40 committed:
doxygen added

arng40 committed:
remove test parent name with constitutive

arng40 committed:
remove unused code + add drawtostring method

arng40 committed:
uncrustify

arng40 committed:
refacto Table first part

arng40 committed:
refortor still going up (missing doc and some minor refacto)

arng40 committed:
uncrustify + perfo example addded

arng40 committed:
update test + margin correction

arng40 committed:
Merge remote-tracking branch 'origin/develop' into feature/dudes/table-layout

arng40 committed:
Merge remote-tracking branch 'origin/feature/dudes/table-layout' into pvt-logs-cp

arng40 committed:
added Formatter function, correction on TableData

arng40 committed:
Delete table..cpp/hpp

arng40 committed:
small correction

arng40 committed:
doc updated

arng40 committed:
TableData2D small refacto + opti

arng40 committed:
first update for log in table function

arng40 committed:
Merge remote-tracking branch 'origin/feature/dudes/table-layout' into pvt-logs-cp

arng40 committed:
uncrustify + small correction

arng40 committed:
csv format / function correction

arng40 committed:
remove dataToString with 2D table parameter

arng40 committed:
code opti

arng40 committed:
add 2D table test

arng40 committed:
uncrustify

arng40 committed:
Alignment::middle by center and compile time verifcation added

arng40 committed:
PR review

arng40 committed:
2nd wave of pr review

arng40 committed:
update compile time verification for table value

arng40 committed:
some update

arng40 committed:
pvt_log updated

arng40 committed:
Merge remote-tracking branch 'origin/feature/dudes/table-layout' into pvt-logs-cp

arng40 committed:
update pvt log

arng40 committed:
add missing doc + uncrustify

arng40 committed:
constness update + refacto on text  formatter

arng40 committed:
PR correction

arng40 committed:
move has_formatter to format.hpp, doc correction

arng40 committed:
log level correction

arng40 committed:
perfo log correctly displayed

arng40 committed:
remove ; to scope bracket

arng40 committed:
doc correction format.hpp

arng40 committed:
format.hpp doc

arng40 committed:
last correction

arng40 committed:
Merge remote-tracking branch 'origin/develop' into feature/dudes/table-layout

arng40 committed:
cmake fix

arng40 committed:
test return string

arng40 committed:
fix hpp

arng40 committed:
fix error with string_view

arng40 committed:
uncrustify

arng40 committed:
Correction melvin PR

arng40 committed:
correction review PR

arng40 committed:
setMargin private

arng40 committed:
uncrustify

arng40 committed:
remove constructor vector string

arng40 committed:
Table Data logs

arng40 committed:
move table to common + pr correction

arng40 committed:
Merge remote-tracking branch 'origin/feature/dudes/table-layout' into pvt-logs-cp

arng40 committed:
correction from merging

arng40 committed:
minor correction

MelReyCG committed:
gracefully breaking your code :)
- make this work
- check which methods can be private
- move every table class to fileIO

MelReyCG committed:
improving proposal + added TODOs

arng40 committed:
Refactor conversion TableData2 => TableData1D

arng40 committed:
Merge remote-tracking branch 'origin/feature/rey/transmissibilityOutput' into feature/dudes/table-layout

arng40 committed:
Merge remote-tracking branch 'origin/develop' into pvt-logs-cp

arng40 committed:
refactor method to convert 2Dtable => 1D table

arng40 committed:
CI correction

arng40 committed:
move table to fileIO

arng40 committed:
update some doc

arng40 committed:
first part test 2D table crash

arng40 committed:
add test/warning for table conversion 2D => 1D and warning for filling table columns from rows

arng40 committed:
ci correction

arng40 committed:
test ci correction for c++ < 17

arng40 committed:
test ci table2D example 1

arng40 committed:
test ci Table2D test 1 & 2

arng40 committed:
test ci Table2D test 1 & 2 & 3

arng40 committed:
add virtual destructor TableFormatter + minor variable refacto

arng40 committed:
bug correction + missing doc + test small  refacto

arng40 committed:
missig doc ...

arng40 committed:
start management logError in table + todo refacto functions in TableFormatter

arng40 committed:
TableFormatter refacto + clean up

arng40 committed:
uncrustify

arng40 committed:
small doc update

arng40 committed:
some  error management update

arng40 committed:
renaming & simplifying

arng40 committed:
add case for fill all column in case of table 2D

arng40 committed:
Merge remote-tracking branch 'origin/develop' into feature/dudes/table-layout

arng40 committed:
simplification

arng40 committed:
code optimization

arng40 committed:
remove useless std::move

arng40 committed:
refacto build separator + reviews

arng40 committed:
final review correction

arng40 committed:
Merge remote-tracking branch 'origin/feature/dudes/table-layout' into pvt-logs-cp

arng40 committed:
method display layout added

arng40 committed:
small correction after merge

arng40 committed:
review correction #1

arng40 committed:
variables renamed

arng40 committed:
uncrustify + doc

arng40 committed:
doc

arng40 committed:
Merge remote-tracking branch 'origin/develop' into feature/dudes/table-layout

arng40 committed:
remove unused code

arng40 committed:
Merge remote-tracking branch 'origin/feature/dudes/table-layout' into pvt-logs-cp

arng40 committed:
syntax correction

arng40 committed:
Merge branch 'pvt-logs-cp' of https://github.com/GEOS-DEV/GEOS into pvt-logs-cp

arng40 committed:
variable correction
